Kitodo-API/src/main/java/org/kitodo/api/dataformat/SortedCollection.java

```diff
@@ -37,104 +37,71 @@
      * @param orderGetter
      *            possibility to get an order number for T
      */
-    public SortedList(Function<T, Integer> orderGetter) {
+    public SortedCollection(Function<T, Integer> orderGetter) {
         this.orderGetter = orderGetter;
     }
 
+    @Override
+    public T get(int index) {
+        throw new UnsupportedOperationException("blocked");
```

Comment:
Why you disallow usage of for loop?

Reply:
I have read your “I assume that only moment when order really matters is iterating over it” comment to the `ensureOrder()` function above that I should sort in `iterator()` only. Which in turn means all other functions that return objects must be blocked, because they would return the wrong order. Maybe I got you wrong. Sorry.

Reply:
I meant: Call sort before the usage of the list, not inside the implementation.